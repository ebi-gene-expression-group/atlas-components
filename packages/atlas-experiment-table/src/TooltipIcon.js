import React from 'react'
import PropTypes from 'prop-types'

const TooltipIcon = ({tooltipText}) =>
  <sup>
    <span
      data-tip={`${tooltipText}`}
      data-html={true}
      className={`icon icon-generic`} data-icon={`i`}
      style={{color: `lightgrey`, fontSize: `small`}}/>
  </sup>

TooltipIcon.propTypes = {
  tooltipText: PropTypes.string.isRequired
}

export default TooltipIcon
