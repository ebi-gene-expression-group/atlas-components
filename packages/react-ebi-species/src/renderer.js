"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
