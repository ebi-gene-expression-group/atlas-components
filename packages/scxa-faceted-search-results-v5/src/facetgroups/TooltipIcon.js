import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import ReactTooltip from 'react-tooltip'

const splitText = str => {
  const maxLineLength = 40

  return str.split(` `).reduce(
    (acc, word) => {
      if (acc[acc.length - 1].length < maxLineLength) {
        acc[acc.length - 1] = `${acc[acc.length - 1]} ${word}`
      } else {
        acc.push(word)
      }
      return acc
    },
    [``]
  ).join(`<br>`).trim()
}

const TooltipIcon = ({ tooltipText }) =>
  <Fragment>
    <sup>
      <span
        data-tip={`<span>${splitText(tooltipText)}</span>`}
        data-html={true}
        className={`icon icon-generic`} data-icon={`i`}
        style={{ color: `lightgrey`, fontSize: `smaller` }}/>
    </sup>
    <ReactTooltip effect={`solid`}/>
  </Fragment>

TooltipIcon.propTypes = {
  tooltipText: PropTypes.string.isRequired
}

export default TooltipIcon
