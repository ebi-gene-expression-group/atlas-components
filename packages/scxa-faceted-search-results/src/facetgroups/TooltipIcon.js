import React from 'react'
import PropTypes from 'prop-types'

const splitText = str => {
  const maxLength = 40

  return str.split(` `).reduce(
    (acc, word) => {
      if (acc[acc.length - 1].length < maxLength) {
        acc[acc.length - 1] = `${acc[acc.length - 1]} ${word}`
      } else {
        acc.push(word)
      }
      return acc
    },
    [``]
  ).join(`<br>`).trim()
}

const TooltipIcon = ({tooltipText}) =>
  <sup>
    <span
      data-tip={`<span>${splitText(tooltipText)}</span>`}
      data-html={true}
      className={`icon icon-generic`} data-icon={`i`}
      style={{color: `lightgrey`, fontSize: `smaller`}}/>
  </sup>

TooltipIcon.propTypes = {
  tooltipText: PropTypes.string.isRequired
}

export default TooltipIcon
