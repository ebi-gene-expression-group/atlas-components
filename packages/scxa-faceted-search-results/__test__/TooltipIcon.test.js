import React from 'react'
import randomWords from 'random-words'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import TooltipIcon from '../src/facetgroups/TooltipIcon'

Enzyme.configure({ adapter: new Adapter() })

describe(`TooltipIcon`, () => {
  test(`breaks very long tooltip texts`, () => {
    const longText = randomWords(100).join(` `)
    const wrapper = shallow(<TooltipIcon tooltipText={longText} />)
    expect(wrapper.find(`span`).prop(`data-tip`).split(`<br>`).length)
      .toBeGreaterThanOrEqual(Math.floor((longText.length / 40) - 1))
  })

  test(`matches snapshot`, () => {
    const longText = `The blowfish puffs himself up four, five times larger than normal and why? Why does he do that? So that it makes him intimidating, that's why. Intimidating! So that the other, scarier fish are scared off. And that's you! You are a blowfish. You see it's just all an illusion. You see it's... it's nothing but air. Now... who messes with the blowfish, Jesse? You're damn right. You are a blowfish. Say it again. Say it like you mean it. You're a BLOWFISH!`
    const tree = renderer.create(<TooltipIcon tooltipText={longText} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
