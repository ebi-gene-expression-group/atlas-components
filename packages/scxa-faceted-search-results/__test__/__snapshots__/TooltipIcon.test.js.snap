// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipIcon matches snapshot 1`] = `
<sup>
  <span
    className="icon icon-generic"
    data-html={true}
    data-icon="i"
    data-tip="<span>The blowfish puffs himself up four, five<br>times larger than normal and why? Why does<br>he do that? So that it makes him intimidating,<br>that's why. Intimidating! So that the other,<br>scarier fish are scared off. And that's you!<br>You are a blowfish. You see it's just all<br>an illusion. You see it's... it's nothing<br>but air. Now... who messes with the blowfish,<br>Jesse? You're damn right. You are a blowfish.<br>Say it again. Say it like you mean it. You're<br>a BLOWFISH!</span>"
    style={
      Object {
        "color": "lightgrey",
        "fontSize": "smaller",
      }
    }
  />
</sup>
`;
